kornia/augmentation/random_generator/base.py

```diff
@@ -17,10 +19,11 @@ def __call__(cls, *args, **kwargs):
 class RandomGeneratorBase(nn.Module, metaclass=_PostInitInjectionMetaClass):
     """Base class for generating random augmentation parameters."""
 
+    device: Optional[Device] = None
+    dtype: torch.dtype
```

Comment:
```suggestion
    dtype: Optional[torch.dtype] = None
```

Reply:
i'll open a ticket to update that

Reply:
Update `RandomGeneratorBase.make_samplers` to have optional `device` and `dtype`

```python
def make_samplers(self, device: Optional[torch.device] = None, dtype: Optional[torch.dtype] = None) -> None:
        raise NotImplementedError
```